### PR TITLE
Fixes 2019: update arch and db docs

### PR DESCRIPTION
**Pull request by jlsherrill**
updates arch docs to include pulp and snapshotting 
update db schema diagram

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-2019

File: docs/architecture.md [hunk 1]
```diff
@@ -4,43 +4,56 @@ Content Sources provides a central location for users to define yum repositories
...
 * Postgresql Database
   * Used for storing our data ([database model](https://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/content-services/content-sources-backend/main/docs/db-model.puml))
 * Kafka Platform
-  * Used to push messages for the event listener and other applications that want information around content sources.  These messages could include Repository creation, update, deletion, etc.
-* Event listener
-  * Will listen for messages via kafka to do background processing tasks.  The results of which are stored in the Postgresql database.
-  * For example, at Repository creation or update time, the listener will fetch metadata from the yum repository and introspect it to learn about its packages.
+  * Used to push messages for notification purposes.  This is no longer used for tasking.
+* Task Worker
+  * Pull tasks from the database and executes them
```
Comment (swadeley) [inline docs/architecture.md:19]:
```suggestion
  * Pulls tasks from the database and executes them  
```

File: docs/architecture.md [hunk 2]
```diff
@@ -4,43 +4,56 @@ Content Sources provides a central location for users to define yum repositories
...
-  * For example, at Repository creation or update time, the listener will fetch metadata from the yum repository and introspect it to learn about its packages.
+  * Used to push messages for notification purposes.  This is no longer used for tasking.
+* Task Worker
+  * Pull tasks from the database and executes them  
+  * For example, at Repository creation or update time, the worker will fetch metadata from the yum repository and introspect it to learn about its packages.
+* Redis cache
+  * Used to cache rbac requeests to decrease latency
+* Pulp Server
+  * Used for creating snapshots of repositories
+  * Consists of a database, api server, content server, and workers
```
Comment (swadeley) [inline docs/architecture.md:25]:
```suggestion
  * Consists of a database, API server, content server, and workers
```

File: docs/architecture.md [hunk 3]
```diff
@@ -4,43 +4,56 @@ Content Sources provides a central location for users to define yum repositories
...
    * Each pod consumes about 30 MB of memory
- * Minimum 3 pods for the Event Listener (kafka consumer)
+ * Minimum 3 pods for the Task Worker
    * Each pod consumes about 35 MB of memory
 
 ## Data Loss
  * Database: If the database is lost, all user data is lost and must be restored from backup.
  * Kafka: If the contents of the kafka queue are lost, a newly created repository will show as 'Pending' and introspection will be delayed until the introspection cron job runs (currently every eight hours).
-
+ * redis: If the content are lost, latency will be higher temporarily until a good cache is built back.
```
Comment (swadeley) [inline docs/architecture.md:42]:
```suggestion
 * redis: If the contents are lost, latency will be higher temporarily until a good cache is built back.
```

File: docs/architecture.md [hunk 4]
```diff
@@ -4,43 +4,56 @@ Content Sources provides a central location for users to define yum repositories
...
+ * Pulp Server: If the pulp database is lost, it must be restored from backup.
+ 
 ## External Dependencies
  * RBAC
    * https://github.com/RedHatInsights/insights-rbac
    * If unavailable, user actions cannot be performed
  * Database
    * If unavailable, user actions cannot be performed, no introspections can take place
+ * Redis
+   * If unavailable, requests to the api will have a higher latency
```
Comment (swadeley) [inline docs/architecture.md:52]:
```suggestion
   * If unavailable, requests to the API will have a higher latency
```

File: docs/architecture.md [hunk 5]
```diff
@@ -4,43 +4,56 @@ Content Sources provides a central location for users to define yum repositories
...
-  * Used to push messages for the event listener and other applications that want information around content sources.  These messages could include Repository creation, update, deletion, etc.
-* Event listener
-  * Will listen for messages via kafka to do background processing tasks.  The results of which are stored in the Postgresql database.
-  * For example, at Repository creation or update time, the listener will fetch metadata from the yum repository and introspect it to learn about its packages.
+  * Used to push messages for notification purposes.  This is no longer used for tasking.
+* Task Worker
+  * Pull tasks from the database and executes them  
+  * For example, at Repository creation or update time, the worker will fetch metadata from the yum repository and introspect it to learn about its packages.
+* Redis cache
+  * Used to cache rbac requeests to decrease latency
```
Comment (swadeley) [inline docs/architecture.md:22]:
```suggestion
  * Used to cache rbac requests to decrease latency
```

**Review by swadeley — COMMENTED** [review]
trivial typos

**Comment by jlsherrill** [comment]
updated

**Review by swadeley — APPROVED** [review]
LGTM

**Pull request merged**